hledger-lib/Hledger/Read/Common.hs

```diff
@@ -677,11 +677,8 @@ priceamountp = option NoPrice $ do
 
 partialbalanceassertionp :: JournalParser m BalanceAssertion
 partialbalanceassertionp = optional $ do
-  sourcepos <- try $ do
-    lift (skipMany spacenonewline)
-    sourcepos <- genericSourcePos <$> lift getSourcePos
-    char '='
-    pure sourcepos
+  sourcepos <- genericSourcePos <$> lift getSourcePos
+  char '='
```

Comment:
Having a hard time understanding this commit.. is it incomplete ? Message needs clarification.

Reply:
Mostly, it doesn't change anything about the behaviour, just the code structure.  These particular lines of code move the skipping of spaces  and the `optional` call out to `JournalReader.hs`, to better reflect that the parse function relates to the assertion itself rather than encoding surrounding file structure as well (which brings it in line with some other parsers in the file).  More generally, that commit just changes `BalanceAssertion` from being a typedef'ed tuple to being a standalone datatype.

Yeah, that could probably be merged with behavioural changes, but this way the foundation is separate from the exact assertions; I could create a new branch off of this commit adding sub-account-inclusive assertions without anything related to commodity-exclusive assertions.

Reply:
Actually, which message were you referring to?  I'm happy enough to try to clarify that, but I'm too close to the code to realize which one seems confusing from the outside.